### DISCUSSION TITLE
@urql/exchange-auth has no types

### DISCUSSION THREAD

**Discussion by ollyde** [urql-graphql/urql#2067] [Q&A] [answered]
Standard Svelte typescript project created with Vite.

`Cannot find module '@urql/exchange-auth' or its corresponding type declarations.`

**Comment by JoviDeCroock** [comment, marked as answer]
It does and points correctly at the file https://www.runpkg.com/?@urql/exchange-auth@0.1.6/package.json#24

**Comment by ollyde** [comment]
@JoviDeCroock thanks for the speedy reply.

Not sure where I'm going wrong here, other packages seem to work just fine :-/

<img width="443" alt="Screenshot 2021-11-08 at 18 11 57" src="https://user-images.githubusercontent.com/6940726/140787189-7573e8c9-1e76-404d-a669-dc860f2dad57.png">

**Reply by JoviDeCroock** [reply]
To follow up to this, I changed it our `vite` example and it works correctly https://github.com/FormidableLabs/urql/blob/main/examples/with-refresh-auth/src/client.js#L10

Same for a new Svelte project

**Comment by ollyde** [comment]
After ```npm install --save @urql/exchange-auth```

**Comment by JoviDeCroock** [comment]
What do you see when you click into the `node_modules`? Do you see the `types` entry for `package.json` and can you follow it to the types file? Maybe there's something wrong in TypeScript or the way we do the pathing

I just did a quick create-react-app and create-next-app and I'm not seeing this

**Comment by ollyde** [comment]
Indeed it looks correct. Hmm.

<img width="772" alt="Screenshot 2021-11-08 at 18 15 04" src="https://user-images.githubusercontent.com/6940726/140787634-b07d8c3c-45b8-4190-9bba-b0d9e6bf6ab2.png">

Perhaps try with Svelte TS; ```npx degit user/project my-project``` ?

**Comment by JoviDeCroock** [comment]
You can leverage `context.fetchOptions` https://formidable.com/open-source/urql/docs/basics/svelte/#context-options

I guess over-engineered is quite subjective as this is how OAuth and refresh-tokens often work in FE-apps 😅

**Comment by JoviDeCroock** [comment]
Those fetchOptions are part of a `store` there, this is kind off going outside of our boundaries of an issue, let's maybe move this to a discussion 😅
